DataFormats/TauReco/interface/PFBaseTau.h

```diff
@@ -0,0 +1,297 @@
+#ifndef DataFormats_TauReco_PFBaseTau_h
+#define DataFormats_TauReco_PFBaseTau_h
+
+/* class PFBaseTau
+ * Based on regular PFTau class, but uses CandidatePtrs instead of 
+ * CandidatePtrs and base refs to reco::Jets instead of refs to reco::Jets
```

Comment:
"uses CandidatePtrs instead of CandidatePtrs" ?

Reply:
Also, you're ending up with two nearly duplicated classes.
Wouldn't it be better to just have one generic class instead of two?

Reply:
Good catch: "uses CandidatePtrs instead of CandidatePtrs" should read as "instead of PFCandidatePtrs" and similarly "instead of refs to reco::PFJets". Concerning the class duplication: one can think on a different design, but the idea was to keep PFTau class (dataformat) untouched. Regardless, I propose to postpone discussion until PR to 100X